### PR TITLE
Add domain navigation email count message

### PR DESCRIPTION
**Pull request by hambai**
#### Changes proposed in this Pull Request

* In the new navigation design we're adding the number of mailboxes/email forwards

Here's how it looks (still in progress so some variations are missing):

Screenshot 2020-04-06 at 17 12 38

Screenshot 2020-04-06 at 17 12 23

Screenshot 2020-04-06 at 17 31 08

Depends on: D41428-code

#### Testing instructions

* Open a domain that has no mailboxes, one with mailboxes and one with email forwards.

### CODE REVIEW

File: client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx [hunk 1]
```diff
@@ -78,17 +79,50 @@ class DomainManagementNavigationEnhanced extends React.Component {
...
+		if ( hasGSuiteWithUs( domain ) ) {
+			const gsuiteMailboxCount = getGSuiteMailboxCount( domain );
+			navigationDescription = translate(
+				'%(gSuiteMailboxCount)d mailbox',
+				'%(gSuiteMailboxCount)d mailboxes',
+				{
+					count: gsuiteMailboxCount,
+					args: {
+						gSuiteMailboxCount: gsuiteMailboxCount,
+					},
```
Comment (stephanethomas) [inline client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx:101]: You can just pass the key if the name of the variable is the same:

```javascript
args: { gSuiteMailboxCount },
```

File: client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx [hunk 2]
```diff
@@ -78,17 +79,50 @@ class DomainManagementNavigationEnhanced extends React.Component {
...
+		const { emailForwardsCount } = domain;
 
 		if ( ! this.isDomainInNormalState() ) {
 			return null;
 		}
 
+		let navigationText;
+		let navigationDescription;
+
+		navigationText = translate( 'Manage your email accounts' );
```
Comment (stephanethomas) [inline client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx:91]: We can shorter this a bit:

```javascript
let navigationText = translate( 'Manage your email accounts' );
```

**Comment by stephanethomas** [comment]
I gave this pull request a try with D41428-code, and everything worked fine. However, I would suggest redirecting to the [`Email Forwarding` page](https://wordpress.com/email/DOMAIN/forwarding/SITE) rather than the [`Email` page](https://wordpress.com/email/DOMAIN/manage/SITE) when a domain only has email forwards (the `Email Forwarding` option is hidden at the bottom of that page, and can easily be missed).

**Comment by stephanethomas** [comment]
> I had a hard time with G Suite though. I think it's because I'm sandboxed. 

@fditrapani: I would avoid sandboxing the store to test G Suite, if that's what you meant. Our tem never does that, and simply use free credits instead. If you still have problems with G Suite, do not hesitate to ping me or @olaseni on Slack though: we'll be happy to help 🙂.

**Comment by hambai** [comment]
> However, I would suggest redirecting to the [`Email Forwarding` page](https://wordpress.com/email/DOMAIN/forwarding/SITE) rather than the [`Email` page](https://wordpress.com/email/DOMAIN/manage/SITE) when a domain only has email forwards (the `Email Forwarding` option is hidden at the bottom of that page, and can easily be missed).

That would actually hide GSuite as a viable option as you would not have any clear path to get GSuite. I agree it's hidden but I believe it was on purpose. One thing we can do is move the Email forwarding navigation menu above GSuite - always or only if you have forwards?

**Comment by fditrapani** [comment]
> I would avoid sandboxing the store to test G Suite, if that's what you meant. Our tem never does that, and simply use free credits instead. 

Thanks for pointing that out @stephanethomas. Don't I need to be sandboxed to apply the patch though? Or is the problem more specifically that I am sandboxing the store as well?

> However, I would suggest redirecting to the Email Forwarding page rather than the Email page when a domain only has email forwards (the Email Forwarding option is hidden at the bottom of that page, and can easily be missed).

You bring up a good point @stephanethomas! and so does @hambai:

> That would actually hide GSuite as a viable option as you would not have any clear path to get GSuite. I agree it's hidden but I believe it was on purpose. One thing we can do is move the Email forwarding navigation menu above GSuite - always or only if you have forwards?

I believe it's on purpose because we want people to pick G Suite. Maybe we should deal with this in a separate PR but another approach could be to take people to the forward screen and present a G Suite up sell on that screen?

**Comment by stephanethomas** [comment]
> Don't I need to be sandboxed to apply the patch though? Or is the problem more specifically that I am sandboxing the store as well?

Yes, and yes 😄. You indeed need to apply the patch to your sandbox, then point `public-api.wordpress.com` to your sandbox. Enabling the store sandbox is a different thing, which basically allows you to use a fake method of payment. You can still purchase G Suite with free credits though, so using the store sandbox is not needed here. I personally avoid using the store sandbox as much as I can as anything purchased with it won't appear in your list of subscriptions when it is disabled (and vice-versa), which I find to be confusing sometimes.

**Comment by stephanethomas** [comment]
> I believe it's on purpose because we want people to pick G Suite. Maybe we should deal with this in a separate PR but another approach could be to take people to the forward screen and present a G Suite up sell on that screen?

I don't have a strong opinion about it, that was just something that I've noticed. Maybe it's better to circle back on this when we start implementing another email provider, as we'll need a better way to present all the different email options to users anyway.

**Pull request merged**